app/src/main/java/fr/neamar/kiss/loader/LoadContactsPojos.java

```diff
@@ -53,8 +54,24 @@ public LoadContactsPojos(Context context) {
                             .getColumnIndex(ContactsContract.Contacts.LOOKUP_KEY));
                     contact.timesContacted = Integer.parseInt(cur.getString(cur
                             .getColumnIndex(ContactsContract.CommonDataKinds.Phone.TIMES_CONTACTED)));
-                    contact.setName(cur.getString(cur
-                            .getColumnIndex(ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME)));
+                    contact.setName(cur.getString(cur.getColumnIndex(ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME)));
+
+                    String contactID = cur.getString(cur.getColumnIndex(ContactsContract.CommonDataKinds.Phone.CONTACT_ID));
+
+                    Cursor cur2 = context.getContentResolver().query(
```

Comment:
I'm concerned with the performance of this implementation -- we're doing a query to the ContactResolver for each contact, and this can be quite long for people with a shitload of contacts (I remember seeing some people use KISS with 10k contacts).

From what I see, I believe it's instead possible to directly load `ContactsContract.CommonDataKinds.Nickname.NAME` from the main query (line :39), do you think this will work?